### DISCUSSION TITLE
Encountered a `504` - `request timed out` error while performing `auth` tasks (e.g., sending recovery or magic links) in a self-hosted Supabase Studio

### DISCUSSION THREAD

**Discussion by Leo-WL** [supabase/supabase#37943] [Questions]
### Discussed in https://github.com/orgs/supabase/discussions/30152

<div type='discussions-op-text'>

<sup>Originally posted by **rafadlis** October 29, 2024</sup>
## Specs

| Spec | detail |
| --- | --- |
| OS | Ubuntu Server |
| Main Tools | aapanel |
| Supabase Deployment | Docker |
| Mail Server | Mail Server aapanel (tested) |

## SMTP config

the SMTP server is running fine but it takes quite a while to send an email.
 
## the logs

Event Message

```json
{
  "auth_event": {
    "action": "user_recovery_requested",
    "actor_id": "03016027-3423-4c7e-b262-6a8cac8e8594",
    "actor_username": "rahmatfadlis@gmail.com",
    "actor_via_sso": false,
    "log_type": "user"
  },
  "component": "api",
  "error": "context deadline exceeded",
  "level": "error",
  "method": "POST",
  "msg": "504: Processing this request timed out, please retry after a moment.",
  "path": "/recover",
  "referer": "http://localhost:3000",
  "remote_addr": "172.19.0.12",
  "request_id": "e180d34f-af64-4e10-acb5-f6fdf2d773bb",
  "time": "2024-10-28T22:47:16Z"
}
```

metadata

```json
[
  {
    "auth_event": [
      {
        "action": "user_recovery_requested",
        "actor_id": "03016027-3423-4c7e-b262-6a8cac8e8594",
        "actor_username": "rahmatfadlis@gmail.com",
        "actor_via_sso": false,
        "log_type": "user"
      }
    ],
    "component": "api",
    "error": "context deadline exceeded",
    "level": "error",
    "method": "POST",
    "msg": "504: Processing this request timed out, please retry after a moment.",
    "path": "/recover",
    "referer": "http://localhost:3000",
    "remote_addr": "172.19.0.12",
    "request_id": "e180d34f-af64-4e10-acb5-f6fdf2d773bb",
    "time": "2024-10-28T22:47:16Z",
    "timestamp": "2024-10-28T22:47:16Z"
  }
]
```

## The Questions

How can I extend the request timeout duration? Are there any alternative solutions to address this issue?</div>

**Comment by Leo-WL** [comment]
I am using a self-hosted email server to send emails. It works when tested on the Supabase official website with this mailer sender(port 587), but it does not function when used in a self-hosted Supabase environment.

And there's no error log in supabase-auth 
{"auth_event":{"action":"user_confirmation_requested","actor_id":"55719a6d-8bba-4035-86e1-040314414ab0","actor_username":"2380377@qq.com","actor_via_sso":false,"log_type":"user","traits":{"provider":"email"}},"component":"api","error":"context deadline exceeded","level":"error","method":"POST","msg":"504: Processing this request timed out, please retry after a moment.","path":"/signup","referer":"http://localhost:3000","remote_addr":"192.9.138.114","request_id":"a25c5565-3e9a-4dbc-92a3-51ea226896e2","time":"2024-11-21T05:27:18Z"}
{"auth_event":{"action":"user_confirmation_requested","actor_id":"55719a6d-8bba-4035-86e1-040314414ab0","actor_username":"2380377@qq.com","actor_via_sso":false,"log_type":"user","traits":{"provider":"email"}},"component":"api","duration":10001339224,"level":"info","method":"POST","msg":"request completed","path":"/signup","referer":"http://localhost:3000","remote_addr":"192.9.138.114","request_id":"a25c5565-3e9a-4dbc-92a3-51ea226896e2","status":504,"time":"2024-11-21T05:27:18Z"}
{"auth_event":{"action":"user_confirmation_requested","actor_id":"55719a6d-8bba-4035-86e1-040314414ab0","actor_username":"2380377@qq.com","actor_via_sso":false,"log_type":"user","traits":{"provider":"email"}},"component":"api","duration":10001339224,"error":"database error on committing or rolling back transaction: sql: transaction has already been committed or rolled back","level":"error","method":"POST","msg":"Unhandled server error: database error on committing or rolling back transaction: sql: transaction has already been committed or rolled back","path":"/signup","referer":"http://localhost:3000","remote_addr":"192.9.138.114","request_id":"a25c5565-3e9a-4dbc-92a3-51ea226896e2","status":504,"time":"2024-11-21T05:27:18Z"}

**Comment by Harsh-WebIntel-Dev** [comment]
anyone resolved this issue?

**Comment by askarov-web2cat** [comment]
У меня была такая ошибка. Оказалось что мой сервис VPS блокировал почтовые порта, 587, 465, 25 и т.д. Я написал в поддержку и они открыли порты которые мне нужны были

**Comment by russoue** [comment]
Sharing in case it helps someone.  I was trying to use Amazon SES and in the beginning tried to use their STARTTLS port 25 and got the same error.  However, after switching to their TLS Wrapper Port 465 the errors went away and it seemed like the emails were being sent.  However, my Amazon SES account is still in sandbox mode and I haven't received the emails I sent to myself yet.  I am guessing that has something to do with the SES account and not GoTrue.

**Comment by Hallidayo** [comment]
Hi all, I'm going to move this over as a troubleshooting discussion. Thanks for your help on this one!
